src/decide.ts

```diff
@@ -121,25 +121,61 @@ async function assessCandidateHelper(
 	if (hashesToExclude.includes(candidateMeta.infoHash)) {
 		return { decision: Decision.INFO_HASH_ALREADY_EXISTS };
 	}
-	const { dataDirs, dataMode } = getRuntimeConfig();
+	const { dataDirs, dataMode, hardlinkDir } = getRuntimeConfig();
 	const perfectMatch = compareFileTrees(candidateMeta, searchee);
 	if (perfectMatch) {
-		return { decision: Decision.MATCH, metafile: candidateMeta };
+		await symlinkExact(searchee.path, hardlinkDir);
+		return { decision: Decision.MATCH, metafile: candidateMeta};
 	}
 	if (!dataDirs || dataDirs.length == 0) {
 		return { decision: Decision.FILE_TREE_MISMATCH };
 	}
 	if (
 		!statSync(searchee.path).isDirectory() &&
 		compareFileTreesIgnoringNames(candidateMeta, searchee) &&
-		dataMode == "risky"
-	) {
-		return {
-			decision: Decision.MATCH_EXCEPT_PARENT_DIR,
-			metafile: candidateMeta,
-		};
+		dataMode == "risky") {
+			if (hardlinkDir) {
+				const candidateParentDir = path.dirname(candidateMeta.files[0].path);
+				var correctedHardlinkDir = hardlinkDir;
+				if (candidateParentDir != ".") {
+					if (!existsSync(path.join(hardlinkDir, candidateParentDir))) {
+						mkdirSync(path.join(hardlinkDir, candidateParentDir));
+					}
+					correctedHardlinkDir = path.join(hardlinkDir, candidateParentDir);
+				}
+				symlinkFile(
+					path.dirname(searchee.path),
+					correctedHardlinkDir,
+					path.basename(searchee.path),
+					path.basename(candidateMeta.files[0].path)
+				);
```

Comment:
i think this code is less related to decisions and more related to action/injection. maybe the top of performAction would be a good place for it? and we could add an argument to `inject` specifying the parent dir